File: src/main/java/github/macro/pieces/Spearman.java
package github.macro.pieces;

import github.macro.Direction;
import github.macro.players.Player;
import org.jetbrains.annotations.NotNull;

import java.util.List;

/**
 * Created by Macro303 on 2018-Feb-12.
 */
public class Spearman extends Piece {
	public Spearman(@NotNull Player player) {
		super(player, 2, "P", List.of(
				Direction.NORTH,
				Direction.EAST,
				Direction.SOUTH,
				Direction.WEST
		));
	}

	@Override
	public String toString() {
		return "Spearman{} " + super.toString();
	}
}

File: src/main/kotlin/github/macro/pieces/Spearman.kt
package github.macro.pieces

import github.macro.Direction
import github.macro.players.Player

/**
 * Created by Macro303 on 2018-02-12.
 */
class Spearman(player: Player) : Piece(
	player = player,
	distance = 2,
	symbol = "\u2694",
	directions = arrayOf(Direction.NORTH, Direction.EAST, Direction.SOUTH, Direction.WEST)
) {
	//	private static final String symbol = "\u2694";
	//	private static final String symbol = "P";
	override fun toString(): String = "Spearman() ${super.toString()}"
}